edb/schema/roles.py

```diff
@@ -72,9 +79,24 @@ def _process_role_body(
     ) -> None:
         password = cmd.get_attribute_value('password')
         if password is not None:
+            if cmd.get_attribute_value('password_hash') is not None:
+                raise errors.EdgeQLSyntaxError(
+                    'cannot specify both `password` and `password_hash` in'
+                    ' the same statement',
+                    context=astnode.context,
+                )
             salted_password = scram.build_verifier(password)
             cmd.set_attribute_value('password', salted_password)
 
+        password_hash = cmd.get_attribute_value('password_hash')
+        if password_hash is not None:
+            if not password_hash.startswith('SCRAM-SHA-256$'):
+                raise errors.InvalidValueError(
+                    'only SCRAM-SHA-256 hashes are supported',
+                    context=astnode.context,
+                )
+            cmd.set_attribute_value('password', password_hash)
```

Comment:
Please add a check that the hash is well-formed. (`scram.parse_verifier` should do the trick).